basic-application/basic-application-webapp/src/main/java/fr/openwide/core/basicapp/web/application/referential/model/CityDataProvider.java

```diff
@@ -0,0 +1,76 @@
+package fr.openwide.core.basicapp.web.application.referential.model;
+
+import org.apache.wicket.injection.Injector;
+import org.apache.wicket.model.IModel;
+import org.apache.wicket.model.Model;
+
+import com.google.common.collect.ImmutableMap;
+
+import fr.openwide.core.basicapp.core.business.referential.model.City;
+import fr.openwide.core.basicapp.core.business.referential.model.search.CitySort;
+import fr.openwide.core.basicapp.core.business.referential.model.search.ICitySearchQuery;
+import fr.openwide.core.jpa.more.business.generic.model.EnabledFilter;
+import fr.openwide.core.jpa.more.business.search.query.ISearchQuery;
+import fr.openwide.core.wicket.more.markup.html.sort.model.CompositeSortModel;
+import fr.openwide.core.wicket.more.markup.html.sort.model.CompositeSortModel.CompositingStrategy;
+import fr.openwide.core.wicket.more.model.AbstractSearchQueryDataProvider;
+import fr.openwide.core.wicket.more.model.GenericEntityModel;
+
+public class CityDataProvider extends AbstractSearchQueryDataProvider<City, CitySort> {
```

Comment:
Couldn't there be an `AbstractGenericListItemDataProvider`, so that data providers for class as simple as City (where we only need searching by "enabledFilter" and by label) would be even more concise?
This would require:
 * adding a `.label(String)` method to `IGenericListItemSearchQuery` (not to the implementation, though)
 * adding the models and getters for those models to `AbstractGenericListItemDataProvider`
 * adding an abstract `createSearchQuery()` method (with no parameter) to `AbstractGenericListItemDataProvider`
 * implementing `getSearchQuery()` in `AbstractGenericListItemDataProvider`

We could even provide an instantiable (and final) `SimpleGenericListItemDataProvider<T, S>`, whose constructor would take as an argument:
 * The actual query class
 * The label sort
 * The ID sort